src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -165,6 +168,46 @@ def get(self, request: Request, project: Project) -> Response:
         return Response(serialize(found_artifacts, request.user))
 
 
+def renew_artifact_bundles(used_artifact_bundles: Mapping[int, datetime]):
+    # We take a snapshot in time that MUST be consistent across all updates.
+    now = datetime.now(tz=pytz.UTC)
+
+    for (bundle_id, date_added) in used_artifact_bundles.items():
+        if ArtifactBundle.can_be_renewed(date_added):
+            # In case this bundle can be renewed we will try to renew all the instances connected to this bundle with
+            # the same `date_added`.
+            for model in (
+                ArtifactBundle,
+                ProjectArtifactBundle,
+                DebugIdArtifactBundle,
+                ReleaseArtifactBundle,
+            ):
+                async_update_model(model=model, bundle_id=bundle_id, new_date_added=now)
+
+
+def async_update_model(model, bundle_id: int, new_date_added: datetime):
+    id_key = "id" if model == ArtifactBundle else "artifact_bundle_id"
+    # We want to asynchronously update the field. This is automatically handled in a batching fashion by the buffer.
+    #
+    # This implementation might have race conditions in which a `buffer_incr` task will be scheduled for the same
+    # `bundle_id` in different times. We assume here that there is some ordering in execution and tasks are completed
+    # in a FIFO fashion without interleaving updates. In case this will turn out to not be the case, we might have
+    # the database into an inconsistent state, where the `date_added` field of the entities connected to the same
+    # `bundle_id` are different.In addition, we don't expect the same `bundle_id` to be request in close sequence since
+    # Symbolicator caches for ~1 hour the bundles received by this endpoint.
```

Comment:
well, symbolicator will cache the unique combination of `release + debug_id + url`, so it will request different urls, and get back the same bundle, depending on the incoming stack trace.

Reply:
Oh right, thanks for the heads up, will change the comment.